### ISSUE TITLE
Introduce separate batch sizes for train, validation, and test

### ISSUE THREAD

**Issue by kdavis-mozilla** [mozilla/DeepSpeech#239]
The distribution of wav lengths in the train, validation, and test sets may be skewed. For example, if small wav files are in the training set and larger ones in the validation set.

If this is the case, then the maximum training batch size is too large for the validation set.

To solve this separate batch sizes should be introduced for the train, validation, and test sets.

Far from being theoretic, this skew exists in the TED data set. The largest wav files form the train, validation, and test sets are as follows:
```
train
-rw-rw-r-- 1 alissy alissy 634K Oct 10 05:47 JonathanHarris_2007P-391.71-411.97.wav

validation
-rw-rw-r-- 1 alissy alissy 1.3M Oct 10 05:47 WadeDavis_2003-123.31-163.23.wav

test
-rw-rw-r-- 1 alissy alissy 1018K Oct 10 05:50 JaneMcGonigal_2010-196.922-229.47.wav
```
As one can see, the largest wav in the validation data set is about twice the size of the largest wav in the test data set. A similar statement is true for the test data set. This causes the batch size 48 to work for the training set and fail for the validation set.
